taskcluster/ci/test/kind.yml

```diff
@@ -8,38 +8,56 @@ transforms:
     - taskgraph.transforms.task:transforms
 
+job-defaults:
+    attributes:
+        build-type: debug
+        code-review: true
+    run:
+        # TODO Generate APKs in a build task instead
+        gradlew: ['clean', 'assembleGeckoNightly', 'assembleAndroidTest', 'assembleDebug']
+        secrets:
+            - name: project/mobile/android-components/firebase
+              key: firebaseToken                  
+              path: .firebase_token.json
+              json: true
+        using: gradlew
+        use-caches: false
+    run-on-tasks-for: [github-push]
+    treeherder:
+            kind: test
+            platform: 'ui-test/opt'
+            tier: 2
+    worker-type: b-android
+    worker:
+        artifacts:
+            - name: public
+              path: /build/android-components/results
+              type: directory
+        docker-image: {in-tree: ui-tests}
+        env:
+            GOOGLE_APPLICATION_CREDENTIALS: '.firebase_token.json'
+            GOOGLE_PROJECT: moz-android-components-230120
+        max-run-time: 7200
+
 jobs:
     ui:
-        attributes:
-            build-type: debug
-            code-review: true
-        description: Test Reference Browser
+        description: 'Run ui tests for browser component'
         run:
-            # TODO Generate APKs in a build task instead
-            gradlew: ['clean', 'assembleGeckoNightly', 'assembleAndroidTest']
             post-gradlew:
                 - ['automation/taskcluster/androidTest/ui-test.sh', 'browser', 'arm', '1']
-            secrets:
-                - name: project/mobile/android-components/firebase
-                  key: firebaseToken
-                  path: .firebase_token.json
-                  json: true
-            using: gradlew
-            use-caches: false
-        run-on-tasks-for: [github-push]
         treeherder:
-            kind: test
-            platform: 'ui-test/opt'
-            symbol: ui
-            tier: 2
-        worker-type: b-android
-        worker:
-            artifacts:
-                - name: public
-                  path: /build/android-components/results
-                  type: directory
-            docker-image: {in-tree: ui-tests}
-            env:
-                GOOGLE_APPLICATION_CREDENTIALS: '.firebase_token.json'
-                GOOGLE_PROJECT: moz-android-components-230120
-            max-run-time: 7200
+            symbol: 'ui-components'
+    ui-browser:
+        description: 'Run ui tests for browser sample'
+        run:
+            post-gradlew:
+                - ['automation/taskcluster/androidTest/ui-test.sh', 'samples-browser', 'arm', '1']
+        treeherder:
+            symbol: 'ui-samples-browser'
+    ui-glean:
+        description: 'Run ui tests for glean sample'
+        run:
+            post-gradlew:
+                - ['automation/taskcluster/androidTest/ui-test.sh', 'samples-glean', 'arm', '1']
+        treeherder:
+            symbol: 'ui-samples-glean'
```

Comment:
So whenever we want to run UI tests for a new component then we just need to add it here? :)

Initially I was wondering whether we could control this via a flag in .buildconfig.yml (boolean), but I'm not sure if they need to be listed separately here or whether some decision task could create them dynamically? Definitely doesn't need to block landing this though. I'm happy about every additional test we run in automation!

@JohanLorenzo What's your opinion here?

Reply:
Yeah, that would be it, either for component or sample, adding the corresponding part there, will create  a  new task and run the tests. 

@pocmo, I know about your thoughts reading the .buildconfig.yml, sorry this is a different approach.  In fact I started by creating a script to read a `ui-tests: boolean` from that file and and getting the apks paths and so. That part is done, but then it got really complicated to select each apk for each task, running them separately on Firebase. It will require more investigation and work, we could do that but I heard that the .buildconfig.yml file may be added to gradle so things around that may change in the future. So while that is decided, we could use this solution and I will keep investigating/working on the original idea. Hope that sounds good :)